x-pack/plugins/upgrade_assistant/public/application/components/overview/fix_logs_step/verify_changes/verify_changes.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent, useState } from 'react';
+import moment from 'moment-timezone';
+import useInterval from 'react-use/lib/useInterval';
+import { FormattedDate, FormattedTime, FormattedMessage } from '@kbn/i18n/react';
+
+import { i18n } from '@kbn/i18n';
+import { EuiCallOut, EuiButton, EuiLoadingContent, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { useAppContext } from '../../../../app_context';
+import { Storage } from '../../../../../shared_imports';
+
+const POLLING_INTERVAL = 60000;
+const LS_SETTING_ID = 'kibana.upgradeAssistant.lastCheckpoint';
+const localStorage = new Storage(window.localStorage);
+
+const i18nTexts = {
+  calloutTitle: (warningsCount: number, previousCheck: string) => (
+    <FormattedMessage
+      id="xpack.upgradeAssistant.overview.verifyChanges.calloutTitle"
+      defaultMessage="{warningsCount, plural, =0 {No} other {{warningsCount}}} deprecation {warningsCount, plural, one {warning} other {warnings}} since {previousCheck}"
+      values={{
+        warningsCount,
+        previousCheck: (
+          <>
+            <FormattedDate value={previousCheck} year="numeric" month="long" day="2-digit" />{' '}
+            <FormattedTime value={previousCheck} timeZoneName="short" hour12={false} />
+          </>
+        ),
+      }}
+    />
+  ),
+  calloutBody: i18n.translate('xpack.upgradeAssistant.overview.verifyChanges.calloutBody', {
+    defaultMessage:
+      'Reset the counter after making changes and continue monitoring to verify that you are no longer using deprecated APIs.',
+  }),
+  loadingError: i18n.translate('xpack.upgradeAssistant.overview.verifyChanges.loadingError', {
+    defaultMessage: 'An error occurred while retrieving the count of deprecation logs',
+  }),
+  retryButton: i18n.translate('xpack.upgradeAssistant.overview.verifyChanges.retryButton', {
+    defaultMessage: 'Try again',
+  }),
+  resetCounterButton: i18n.translate(
+    'xpack.upgradeAssistant.overview.verifyChanges.resetCounterButton',
+    {
+      defaultMessage: 'Reset counter',
+    }
+  ),
+};
+
+const getPreviousCheckpointDate = () => {
+  const storedValue = moment(localStorage.get(LS_SETTING_ID));
+
+  if (storedValue.isValid()) {
+    return storedValue.toISOString();
+  }
+
+  const now = moment().toISOString();
+  localStorage.set(LS_SETTING_ID, now);
+
+  return now;
+};
+
+export const VerifyChanges: FunctionComponent = () => {
```

Comment:
[nit] Can we rename this component and filename to be a little more descriptive?